ref(discover): support saving interval

wmak opened the pull request:
- This adds interval as a field to the discover query so the user selected query can be persisted

wmak committed:
ref: support saving interval

wmak committed:
ref: Validate the interval

narsaynorath commented on src/sentry/discover/endpoints/serializers.py:
```diff
@@ -209,6 +212,15 @@ def validate(self, data):
         if "query" in query:
+            if "interval" in query:
+                interval = parse_stats_period(query["interval"])
+                date_range = self.context["params"]["end"] - self.context["params"]["start"]
```
Should we be using the helper `get_date_range_from_params` here to get the start and end dates based off of `range`? I added a breakpoint in test that's failing and the date_range here spans ~3months

wmak replied:
not quite, that should've been already called at this point, turns out we were only checking `request.GET` and not the post data so it wasn't actually doing anything. Fixed that here
66a6691303949f608f5e1853a3b0f36f80686b59

wmak committed:
fix: Check both request.GET and request.data

- This fixes a bug where get_date_range_from_params wasn't working cause
  we were only checking the GET, but this function is also used in POSTs
  for validation

wmak committed:
fix: typing

wmak committed:
fix: check data exists

wmak committed:
fix: range not statsPeriod hmmm
